x-pack/plugins/infra/public/components/log_stream/log_stream.stories.mdx

```diff
@@ -0,0 +1,349 @@
+import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
+import { Subject } from 'rxjs';
+
+import { I18nProvider } from '@kbn/i18n/react';
+import { EuiThemeProvider } from '../../../../observability/public';
+import { KibanaContextProvider } from '../../../../../../src/plugins/kibana_react/public';
+
+import { DEFAULT_SOURCE_CONFIGURATION } from '../../test_utils/source_configuration';
+import { generateFakeEntries, ENTRIES_EMPTY } from '../../test_utils/entries';
+
+import { LogStream } from './';
+
+<!-- Prework -->
+
+export const startTimestamp = 1595145600000;
+export const endTimestamp = startTimestamp + 15 * 60 * 1000;
+
+export const fetch = function (url, params) {
+  switch (url) {
+    case '/api/infra/log_source_configurations/default':
+      return DEFAULT_SOURCE_CONFIGURATION;
+    case '/api/log_entries/entries':
+      const body = JSON.parse(params.body);
+      if (body.after?.time === body.endTimestamp || body.before?.time === body.startTimestamp) {
+        return ENTRIES_EMPTY;
+      } else {
+        const entries = generateFakeEntries(
+          200,
+          body.startTimestamp,
+          body.endTimestamp,
+          body.columns || DEFAULT_SOURCE_CONFIGURATION.data.configuration.logColumns
```

Comment:
Nit: Would this be more accurate?

```suggestion
          body.columns ?? DEFAULT_SOURCE_CONFIGURATION.data.configuration.logColumns
```

Reply:
In this case they are equivalent. `columns` is either nothing or an array.